libs/interpret-vision/src/lib/VisionExplanationDashboard/Controls/DataCharacteristics.tsx

```diff
@@ -0,0 +1,348 @@
+                        <List
+                          items={list}
+                          onRenderCell={this.onRenderCell}
+                          getPageHeight={this.getPageHeight}
+                          getItemCountForPage={this.getItemCountForPage}
+                          startIndex={this.state.renderStartIndex[index]}
+                        />
+                      </Stack.Item>
+                      <Stack.Item
+                        className={classNames.iconContainer}
+                        onClick={this.loadNextItems(index)}
+                      >
+                        <Icon iconName="DoubleChevronRight12" />
+                      </Stack.Item>
+                    </Stack>
+                  )}
+                </Stack>
+              );
+            })}
+          </Stack.Item>
+        </Stack>
+      </FocusZone>
+    );
+  }
+
+  private onRenderCell = (item?: IVisionListItem | undefined) => {
+    const classNames = dataCharacteristicsStyles();
+
+    if (!item) {
+      return <div />;
+    }
+
+    return (
+      <Stack className={classNames.tile}>
+        <Stack.Item
+          style={{ height: this.props.imageDim, width: this.props.imageDim }}
+        >
+          <Image
+            alt={item?.predictedY}
+            src={`data:image/jpg;base64,${item?.image}`}
+            onClick={this.callbackWrapper(item)}
+            style={{
+              height: "auto",
+              left: 0,
+              position: "relative",
+              top: 0,
+              width: this.props.imageDim
+            }}
+          />
+        </Stack.Item>
+        <div
+          className={
+            item.predictedY === item.trueY
+              ? classNames.successIndicator
+              : classNames.errorIndicator
+          }
+          style={{ marginBottom: 20, width: this.props.imageDim }}
+        />
+      </Stack>
+    );
+  };
+
+  private loadNextItems = (index: number) => () => {
+    const renderStartIndex = this.state.renderStartIndex;
+    renderStartIndex[index] += this.columnCount;
+    this.setState({ renderStartIndex });
+  };
+
+  private callbackWrapper = (item: IVisionListItem) => () => {
+    this.props.selectItem(item);
+  };
+
+  private getPageHeight = () => {
+    return this.rowHeight;
+  };
+
+  private getItemCountForPage = (
+    itemIndex?: number | undefined,
+    visibleRect?: IRectangle | undefined
+  ): number => {
+    if (!visibleRect) {
+      return this.columnCount;
+    }
+    if (itemIndex === 0) {
+      this.columnCount = Math.ceil(visibleRect.width / this.props.imageDim) - 1;
+      this.rowHeight = Math.floor(visibleRect.width / this.columnCount);
+    }
+    return this.columnCount;
+  };
+
+  private showAll = () => {
+    const visibilities = this.state.labelVisibilities;
+
+    for (const label of visibilities.keys()) {
+      visibilities.set(label, true);
+    }
+    this.setState({ labelVisibilities: visibilities });
+  };
+}
```

Comment:
reduce file size to be under 300 lines by breaking this up into separate components